### PR TITLE
Fix ingress api checks

### LINKED ISSUE #305: Incorrect check for Ingress APIversion for sonarqube 9.8.0 and 9.6.6
For both sonarqube charts, there is an error checking for APIversion capabilities:
```helm
{{- if .Capabilities.APIVersions.Has "networking.k8s.io/v1/Ingress" -}}
apiVersion: networking.k8s.io/v1
{{- else if .Capabilities.APIVersions.Has "networking.k8s.io/v1beta1/Ingress" -}}
apiVersion: networking.k8s.io/v1beta1
{{- else -}}
apiVersion: extensions/v1beta1
{{- end }}
```

`/Ingress` needs to be dropped, as that is part of `Kind:`.

So it would be something like this:
```helm
{{- if .Capabilities.APIVersions.Has "networking.k8s.io/v1" -}}
apiVersion: networking.k8s.io/v1
{{- else if .Capabilities.APIVersions.Has "networking.k8s.io/v1beta1" -}}
apiVersion: networking.k8s.io/v1beta1
{{- else -}}
apiVersion: extensions/v1beta1
{{- end }}
```

As it is now, both charts fail on deployment.

### PR DESCRIPTION
**Pull request by rjkernick**
fixes #305

### CODE REVIEW

File: charts/sonarqube/templates/ingress.yaml [hunk 1]
```diff
@@ -1,9 +1,9 @@
 {{- if .Values.ingress.enabled -}}
 {{- $serviceName := include "sonarqube.fullname" . -}}
 {{- $servicePort := .Values.service.externalPort -}}
-{{- if .Capabilities.APIVersions.Has "networking.k8s.io/v1/Ingress" -}}
+{{- if .Capabilities.APIVersions.Has "networking.k8s.io/v1" -}}
```
Comment (EvgeniGordeev) [inline charts/sonarqube/templates/ingress.yaml:4]: the same check on line 47 is required.

File: charts/sonatype-nexus/templates/ingress.yaml [hunk 2]
```diff
@@ -10,9 +10,9 @@
   {{- $servicePort = .Values.nexusProxy.port -}}
 {{- end }}
 
-{{- if .Capabilities.APIVersions.Has "networking.k8s.io/v1/Ingress" -}}
+{{- if .Capabilities.APIVersions.Has "networking.k8s.io/v1" -}}
```
Comment (EvgeniGordeev) [inline charts/sonatype-nexus/templates/ingress.yaml:13]: see line 43.

**Review by EvgeniGordeev — CHANGES_REQUESTED** [review]
missed check on paths array.

**Comment by Mikaciu** [comment]
Hello, 

Shouldn't line 47 of the SonarQube ingress have the `$` context selector ?
https://github.com/Oteemo/charts/blob/85203e9b21e180d7b8d6a3872dd20ca05a2122d6/charts/sonarqube/templates/ingress.yaml#L47 

Suggestion :
```suggestion
-             {{- if .Capabilities.APIVersions.Has "networking.k8s.io/v1/Ingress" }}
+            {{- if $.Capabilities.APIVersions.Has "networking.k8s.io/v1/Ingress" }}
```

HTH

File: charts/sonatype-nexus/templates/ingress.yaml (hunk 2 above)
Comment (rjkernick) [reply charts/sonatype-nexus/templates/ingress.yaml:13]: Thanks for the catch. Fixed

**Comment by rjkernick** [comment]
@Mikaciu I do not believe you need the $

**Comment by EvgeniGordeev** [comment]
> @Mikaciu I do not believe you need the $

Error: UPGRADE FAILED: template: sonar/charts/sonarqube/templates/ingress.yaml:47:32: executing "sonar/charts/sonarqube/templates/ingress.yaml" at <.Capabilities.APIVersions.Has>: nil pointer evaluating interface {}.APIVersions

it must be ```{{- if $.Capabilities.APIVersions.Has "networking.k8s.io/v1" }}``` as @Mikaciu pointed out.

**Comment by patsevanton** [comment]
After this commit broken ingress for kubernetes 1.18
https://github.com/Oteemo/charts/issues/311

**Comment by patsevanton** [comment]
**kubectl version** 
```
Server Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.20",
```

**helm install -f my-values.yaml sonatype-nexus oteemocharts/sonatype-nexus --version 5.3.3**
```
Error: unable to build kubernetes objects from release manifest: unable to recognize "": no matches for kind "Ingress" in version "networking.k8s.io/v1"
```

**helm install --atomic -f my-values.yaml sonatype-nexus oteemocharts/sonatype-nexus --version 5.3.2**
```
NAME: sonatype-nexus
LAST DEPLOYED: Mon Sep 20 17:17:45 2021
NAMESPACE: default
STATUS: deployed
REVISION: 1
TEST SUITE: None
NOTES:
- To access Nexus:

  NOTE: It may take a few minutes for the ingress load balancer to become available or the backends to become HEALTHY.
        You can watch the status of the backends by running:
        `kubectl get ingress -o jsonpath='{.items[*].metadata.annotations.ingress\.kubernetes\.io/backends}'`

  To access Nexus you can check:
   http://nexus.xxxxx.sslip.io

- Login with the following credentials

   username: admin
   password: xxxx
- Next steps in configuration

   Please follow the link below to the README for nexus configuration, usage, backups and DR info:
   https://github.com/Oteemo/charts/tree/master/charts/sonatype-nexus#after-installing-the-chart
```

**Comment by patsevanton** [comment]
**kubectl version**
```
Server Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.10", 
```

**helm install --atomic -f my-values.yaml sonatype-nexus oteemocharts/sonatype-nexus --version 5.3.3**
```
NAME: sonatype-nexus
LAST DEPLOYED: Mon Sep 20 17:18:04 2021
NAMESPACE: default
STATUS: deployed
REVISION: 1
TEST SUITE: None
NOTES:
- To access Nexus:

  NOTE: It may take a few minutes for the ingress load balancer to become available or the backends to become HEALTHY.
        You can watch the status of the backends by running:
        `kubectl get ingress -o jsonpath='{.items[*].metadata.annotations.ingress\.kubernetes\.io/backends}'`

  To access Nexus you can check:
   http://nexus.xxxxxx.sslip.io

- Login with the following credentials

   username: admin
   password: xxxxx
- Next steps in configuration

   Please follow the link below to the README for nexus configuration, usage, backups and DR info:
   https://github.com/Oteemo/charts/tree/master/charts/sonatype-nexus#after-installing-the-chart
```


**helm install --atomic -f my-values.yaml sonatype-nexus oteemocharts/sonatype-nexus --version 5.3.2**
```
NAME: sonatype-nexus
LAST DEPLOYED: Mon Sep 20 17:22:38 2021
NAMESPACE: default
STATUS: deployed
REVISION: 1
TEST SUITE: None
NOTES:
- To access Nexus:

  NOTE: It may take a few minutes for the ingress load balancer to become available or the backends to become HEALTHY.
        You can watch the status of the backends by running:
        `kubectl get ingress -o jsonpath='{.items[*].metadata.annotations.ingress\.kubernetes\.io/backends}'`

  To access Nexus you can check:
   http://nexus.84-252-132-48.sslip.io

- Login with the following credentials

   username: admin
   password: admin123
- Next steps in configuration

   Please follow the link below to the README for nexus configuration, usage, backups and DR info:
   https://github.com/Oteemo/charts/tree/master/charts/sonatype-nexus#after-installing-the-chart
```

**Pull request merged**